src/main/java/org/elasticsearch/index/gateway/local/LocalIndexShardGateway.java

```diff
@@ -162,27 +162,18 @@ public void recover(boolean indexShouldExists, RecoveryState recoveryState) thro
                 throw new IndexShardGatewayRecoveryException(shardId(), "failed to fetch index version after copying it over", e);
             }
             recoveryState.getIndex().updateVersion(version);
-            recoveryState.getIndex().time(System.currentTimeMillis() - recoveryState.getIndex().startTime());
+            recoveryState.getIndex().stopTime(System.currentTimeMillis());
 
             // since we recover from local, just fill the files and size
             try {
-                int numberOfFiles = 0;
-                long totalSizeInBytes = 0;
+                RecoveryState.Index index = recoveryState.getIndex();
                 for (String name : indexShard.store().directory().listAll()) {
```

Comment:
yeah. I tend to work on 1.x when it's non-trivial in terms of BWC. I'll rebase once we're done.